bpo-30871: Add test.pythoninfo

vstinner opened the pull request:
Add Lib/test/pythoninfo.py: script collecting various informations
about Python to help debugging test failures.

<!-- issue-number: bpo-30871 -->
https://bugs.python.org/issue30871
<!-- /issue-number -->

vstinner committed:
bpo-30871: Add test.pythoninfo

* Add Lib/test/pythoninfo.py: script collecting various informations
  about Python to help debugging test failures.
* regrtest: remove sys.hash_info and sys.flags from header.
* Travis CI, Appveyor: run pythoninfo before tests

vstinner commented:
@zware: Would you mind to check if I added the pythoninfo run to all ways to run Python tests on all our CIs?

vstinner commented:
Oh, AppVeyor doesn't want to run on my PR? I'm not sure about my PCbuild/rt.bat change.

zware commented:
If you're only going for CI runs, then it should be removed from `PCbuild\rt.bat` and added to `Tools\buildbot\test.bat` instead.  If you're going for all test runs, it could be removed from `make buildbottest` and added to `Tools\scripts\run_test.py` instead.

I think my actual preference would be to add a CLI option to regrtest, and add that flag to `make buildbottest` and `Tools\buildbot\test.bat`.

Also, AppVeyor isn't running because the most recent commit only changes .travis.yml, which AppVeyor ignores.

JulienPalard commented on Lib/test/pythoninfo.py:
```diff
@@ -0,0 +1,298 @@
+        if key in self.info:
+            raise ValueError("duplicate key: %r" % key)
+
+        # accepted types: str, tuple of str, None
```
tuples of str does not looks accepted here, the following TypeError looks more in-sync with the implementation "value type must be str, int or None". Also I'd place this kind of comment in the docstring, it's not that far and may be usefull?

vstinner replied:
comment removed

JulienPalard commented on Lib/test/pythoninfo.py:
```diff
@@ -0,0 +1,298 @@
+
+    def get_infos(self):
+        """
+        Get informations at a key:value dictionary where values are strings.
```
s/at/as/

vstinner replied:
fixed

vstinner commented:
@zware: "If you're only going for CI runs, then it should be removed from PCbuild\rt.bat and added to Tools\buildbot\test.bat instead. If you're going for all test runs, it could be removed from make buildbottest and added to Tools\scripts\run_test.py instead. I think my actual preference would be to add a CLI option to regrtest, and add that flag to make buildbottest and Tools\buildbot\test.bat."

I don't want to pollute test output with pythoninfo. My plan is to run pythoninfo as a separated command to be able to collapse its output.

In AppVeyor and Travis CI config, I added the pythoninfo run in the build step, so it may be collapsed by default.

"Also, AppVeyor isn't running because the most recent commit only changes .travis.yml, which AppVeyor ignores."

Oh ok. I rebased my patch serie into a single commit and rebased it, so AppVeyor will run.

vstinner commented:
@zware: Oh, I forgot to mention that I plan to add a new "pythoninfo" step on buildbots once pythoninfo is added to 2.7, 3.6 and master branches.

zware reviewed (approved):
I've not reviewed the script itself in detail, but the CI changes LGTM.
